engine/components-api/src/main/java/com/cloud/template/TemplateManager.java

```diff
@@ -56,6 +56,13 @@ public interface TemplateManager {
             + "will validate if the provided URL is resolvable during the register of templates/ISOs before persisting them in the database.",
             true);
 
+    ConfigKey<Boolean> TemplateDeleteFromPrimaryStorage = new ConfigKey<Boolean>("Advanced",
```

Comment:
@abh1sar do we really need a global setting for this usecase, I think we can directly delete the template if there is no usage found.